bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -501,7 +503,7 @@ public int getTimeSecond() {
      */
     @Deprecated
     public long getTime() {
-        return timestamp * 1000L;
+        return ((long)timestamp & 0xFFFFFFFFL) * 1000L;
```

Comment:
The current regression test for this method isn't good enough to ensure that this cast + mask is correct, since when I revert this method to its previous implementation all the tests still pass. Best to add some more tests that exercise the new behavior.

Reply:
I have added tests for `getTime` for the integer corner cases.